### PR TITLE
#3274 cleanup authn service iam v1 vs v2

### PR DESCRIPTION
**Pull request by srenatus**
### :nut_and_bolt: Description: What code changed, and why?

* remove ResetToV1 gRPC method
* fix utterly broken `migrate-tokens` script
* fix error reporting for CreateTokenWithValue when the provided value is too short

Please see the individual commits. 

### :chains: Related Resources

#3274 

### :+1: Definition of Done

Stuff still works. (More stuff works than worked before). Less code. ;)

### :athletic_shoe: How to Build and Test the Change

For the token migrations:

    [61][default:/src/components/authn-service:113]# export TOK=$(chef-automate iam token create --admin token)
    [62][default:/src/components/authn-service:113]# chef-automate dev grpcurl secrets-service -- -d '{"name":"supertoken", "tags": [ {"key": "active", "value": "1"}, {"key": "namespace", "value":  "clientTokens"}, {"key": "token", "value": "secret"}]}'  chef.automate.api.secrets.SecretsService.Create
    {
      "id": "1c691567-1c5a-4b2d-a0b5-92418216e667"
    }
    [63][default:/src/components/authn-service:0]# chef-automate dev grpcurl secrets-service -- chef.automate.api.secrets.SecretsService.List
    {
      "secrets": [
        {
          "id": "23fbf948-9e8a-4790-81fa-868a70620264",
          "name": "asdf",
          "type": "ssh",
          "lastModified": "2020-04-20T13:43:33Z"
        },
        {
          "id": "1c691567-1c5a-4b2d-a0b5-92418216e667",
          "name": "supertoken",
          "lastModified": "2020-04-20T13:52:27Z",
          "tags": [
            {
              "key": "active",
              "value": "1"
            },
            {
              "key": "namespace",
              "value": "clientTokens"
            },
            {
              "key": "token",
              "value": "secret"
            }
          ]
        }
      ],
      "total": 2
    }
    [64][default:/src/components/authn-service:0]# bin/migrate-tokens --auth-token $TOK
    migrating tokens from secrets service to authz service using:
            secrets URL: https://localhost/api/v0/secrets/search
            tokens URL: https://localhost/apis/iam/v2/tokens
            api token: 3C1z0rcVzsyO2zwQSwJmIkf48Kc=

    tokens read from secrets service...
            reading token 'supertoken' from secrets
            total tokens read from secrets service: 1

    writing tokens to authz service...
            wrote token "supertoken" successfully
            total tokens written to authz service: 1
    [65][default:/src/components/authn-service:0]# chef-automate dev grpcurl authn-service -- chef.automate.domain.authn.TokensMgmt/GetTokens
    {
      "tokens": [
        {
          "id": "token",
          "name": "token",
          "value": "3C1z0rcVzsyO2zwQSwJmIkf48Kc=",
          "active": true,
          "created": "2020-04-20T11:21:12Z",
          "updated": "2020-04-20T11:21:12Z"
        },
        {
          "id": "supertoken",
          "name": "supertoken",
          "value": "secretsecretsecretsecret",
          "active": true,
          "created": "2020-04-20T13:59:18Z",
          "updated": "2020-04-20T13:59:18Z"
        }
      ]
    }
    [66][default:/src/components/authn-service:0]#

...where `bin/migrate-tokens` is built from running `make migrate` in `components/authn-service` (it's also shipped with the hart of you rebuild `authn-service`).

### :white_check_mark: Checklist

- [ ] I have read the [CONTRIBUTING document](https://github.com/chef/automate/blob/master/CONTRIBUTING.md).
- [ ] Tests added/updated?
- [ ] Docs added/updated?
- [ ] All commits have been signed-off for the [Developer Certification of Origin](https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco).

### CODE REVIEW

File: components/authn-service/cmd/migrate-tokens/migrate-tokens.go [hunk 1]
```diff
@@ -15,7 +15,7 @@ var (
 )
 
 const (
-	tokensURL  = "https://localhost/iam/v2/tokens"         // nolint
+	tokensURL  = "https://localhost/apis/iam/v2/tokens"    // nolint
```
Comment (bcmdarroch) [inline components/authn-service/cmd/migrate-tokens/migrate-tokens.go:18]: woof 😅 hope no one ran into this

File: components/automate-gateway/handler/iam/v2/tokens/tokens.go [hunk 2]
```diff
@@ -116,16 +116,6 @@ func (s *Server) ListTokens(
 	return &pb_resp.ListTokensResp{Tokens: tokens}, nil
 }
 
-// ResetAllTokenProjects removes all projects from every token
-func (s *Server) ResetAllTokenProjects(
```
Comment (bcmdarroch) [inline components/automate-gateway/handler/iam/v2/tokens/tokens.go:120]: yeah this is fine to delete. it was only used to clear out any straggling `default` project references

File: components/authn-service/tokens/util/tokens.go [hunk 3]
```diff
@@ -21,7 +21,7 @@ func MinimumTokenLength() int {
 func IsValidLegacyToken(token string) error {
 	if len(token) < MinimumLegacyTokenLength {
 		return fmt.Errorf(
-			"legacy token %q is invalid; it is length %d but must be at least %d characters",
+			"legacy token %q is invalid; it has length %d but must be at least %d characters",
```
Comment (bcmdarroch) [inline components/authn-service/tokens/util/tokens.go:24]: 🤓

File: components/authn-service/cmd/migrate-tokens/migrate-tokens.go (hunk 1 above)
Comment (srenatus) [reply components/authn-service/cmd/migrate-tokens/migrate-tokens.go:18]: I think (no data) this is very unlikely.

File: components/authn-service/tokens/pg/adapter.go [hunk 4]
```diff
@@ -32,12 +32,12 @@ func (a *adapter) CreateToken(ctx context.Context,
...
 	id, value, name string, active bool, projects []string) (*tokens.Token, error) {
 	if err := tutil.IsValidToken(value); err != nil {
-		return nil, err
+		return nil, status.Error(codes.InvalidArgument, err.Error())
 	}
 
 	err := a.validateTokenInputs(ctx, name, []string{}, projects, false)
 	if err != nil {
-		return nil, err
+		return nil, err // already a grpc status
```
Comment (blakestier) [inline components/authn-service/tokens/pg/adapter.go:40]: thanks for the comment. Also, this speaks to our validation inconsistencies . . .

File: components/authn-service/tokens/util/tokens.go (hunk 3 above)
Comment (blakestier) [reply components/authn-service/tokens/util/tokens.go:24]: I mean if we're fixing it maybe "it has X characters should but be at least Y characters" but also w/e

**Comment by srenatus** [comment]
> Not clear how we would have gotten a successful result with that...?

Oh we didn't! The console log was doctored, in between I've found that the error was swallowed, fixed it, etc. etc. I'll fix the preamble!

**Comment by srenatus** [comment]
will merge when green

**Pull request merged**